lib/httparty/request.rb

```diff
@@ -276,7 +276,7 @@ def handle_response(body, &block)
           unless options[:maintain_method_across_redirects] && options[:resend_on_redirect]
             self.http_method = Net::HTTP::Get
           end
-        else
+        elsif last_response.code.to_s != '307' && last_response.code.to_s != '308'
```

Comment:
When I was testing, the type of `code` seemed to be inconsistent. I could try undoing that and seeing if it makes any difference.